frontend/src/components/note/NoteModal.vue

```diff
@@ -1,43 +1,46 @@
 <script setup lang="ts">
 import { Form } from 'vee-validate';
+import { ref } from 'vue';
 import { mixed, object, string } from 'yup';
 
 import { Calendar, Dropdown, InputText, TextArea } from '@/components/form';
 import { Button, Dialog, useConfirm, useToast } from '@/lib/primevue';
 import { noteService } from '@/services';
+import { useSubmissionStore } from '@/store';
 import { BringForwardTypes, NoteTypes } from '@/utils/constants';
 import { BRING_FORWARD_TYPES, NOTE_TYPES } from '@/utils/enums';
 
-import type { Note } from '@/types';
-import { nextTick, ref, watch } from 'vue';
 import type { Ref } from 'vue';
+import type { Note } from '@/types';
 
 // Props
 type Props = {
+  activityId: string;
   note?: Note;
-  activityId?: string;
 };
 
 const props = withDefaults(defineProps<Props>(), {
-  note: undefined,
-  activityId: undefined
+  note: undefined
 });
 
-// Emits
-const emit = defineEmits(['note:delete', 'note:submit']);
+// Store
+const submissionStore = useSubmissionStore();
 
 // State
-const visible = defineModel<boolean>('visible');
 const formRef: Ref<InstanceType<typeof Form> | null> = ref(null);
-const showBringForward: Ref<boolean> = ref(false);
+const showBringForward: Ref<boolean> = ref(props.note?.noteType === NOTE_TYPES.BRING_FORWARD);
+const visible = defineModel<boolean>('visible');
 
 // Default form values
 let initialFormValues: any = {
-  createdAt: new Date(),
-  bringForwardDate: null,
-  bringForwardState: null,
+  activityId: props.note?.activityId,
+  bringForwardDate: props.note?.bringForwardDate ? new Date(props.note.bringForwardDate) : null,
+  bringForwardState: props.note?.bringForwardState ?? null,
+  createdAt: props.note?.createdAt ? new Date(props.note.createdAt) : null,
```

Comment:
If createdAt defaults to null with the date field disabled from editing on new notes, no new notes can be created as it trips up the backend validation. Validation expects a createdAt date from the frontend.

Reply:
Using `new Date` solved the new note creation issue.

The new date is now established when the component loads and is set. So if the component was loaded at 10:45am and the navigator does a few things without navigating away and then makes a new note at 10:50am, every single note created after is still stuck at 10:45am.